main.go

```diff
@@ -317,8 +283,77 @@ func main() {
 	}
 }
 
-func before(c *cli.Context) (*state.HelmState, helmexec.Interface, error) {
-	file := c.GlobalString("file")
+func eachDesiredStateDo(c *cli.Context, converge func(*state.HelmState, helmexec.Interface) []error) error {
+	fileOrDirPath := c.GlobalString("file")
+	desiredStateFiles, err := findDesiredStateFiles(fileOrDirPath)
+	if err != nil {
+		return err
+	}
+	for _, f := range desiredStateFiles {
+		state, helm, err := loadDesiredStateFromFile(c, f)
+		if err != nil {
+			return err
+		}
+		errs := converge(state, helm)
+		if err := clean(state, errs); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func findDesiredStateFiles(fileOrDirPath string) ([]string, error) {
+	var dir string
+
+	if fileOrDirPath != "" {
+		if !fileExists(fileOrDirPath) {
+			return []string{}, fmt.Errorf("state file named %s is not found", fileOrDirPath)
+		}
+
+		if !directoryExists(fileOrDirPath) {
+			return []string{fileOrDirPath}, nil
+		}
+
+		dir = fileOrDirPath
+	}
+
+	if dir == "" && fileExists(DefaultHelmfileDirectory) && directoryExists(DefaultHelmfileDirectory) {
+		dir = DefaultHelmfileDirectory
+	}
+
+	if dir != "" {
```

Comment:
I think this is clearer
```
if dir != "" { 
...
else if  fileExists(DefaultHelmfileDirectory) && directoryExists(DefaultHelmfileDirectory) {
...
}
```

Reply:
What would be included in `if` and `else if` bodies respectively? We need `dir` to be set for use in https://github.com/roboll/helmfile/pull/160/files#diff-7ddfb3e035b42cd70649cc33393fe32cR325.

Reply:
Would it look clearer now?